com.unity.render-pipelines.core/ShaderLibrary/HDROutput.hlsl

```diff
@@ -0,0 +1,657 @@
+    outLinear.y = PQToLinear(value.y);
+    outLinear.z = PQToLinear(value.z);
+    return outLinear;
+}
+
+
+// --------------------------------------------------------------------------------------------
+
+// --------------------------------
+//  Color Space transforms
+// --------------------------------
+// As any other space transform, changing color space involves a change of basis and therefore a matrix multiplication.
+// Note that Rec2020 and Rec2100 share the same color space.
+
+float3 RotateRec709ToRec2020(float3 Rec709Input)
+{
+    static const float3x3 Rec709ToRec2020Mat = float3x3(
+
+        0.627402, 0.329292, 0.043306,
+        0.069095, 0.919544, 0.011360,
+        0.016394, 0.088028, 0.895578
+    );
+
+    return mul(Rec709ToRec2020Mat, Rec709Input);
+}
+
+float3 RotateRec2020ToRec709(float3 Rec2020Input)
+{
+    static const float3x3 Rec2020ToRec709Mat = float3x3(
+         1.660496, -0.587656, -0.072840,
+        -0.124547,  1.132895, -0.008348,
+        -0.018154, -0.100597,  1.118751
+    );
+    return mul(Rec2020ToRec709Mat, Rec2020Input);
+}
+
+float3 RotateRec709ToOutputSpace(float3 Rec709Input)
+{
+#ifdef HDR_OUTPUT_SCRGB
+    return Rec709Input;
+#else
+    return RotateRec709ToRec2020(Rec709Input);
+#endif
+}
+
+float3 RotateRec2020ToOutputSpace(float3 Rec2020Input)
+{
+#ifdef HDR_OUTPUT_SCRGB
+    return RotateRec2020ToRec709(Rec2020Input);
+#else
+    return Rec2020Input;
+#endif
+}
+
+float3 RotateRec2020ToLMS(float3 Rec2020Input)
+{
+    static const float3x3 Rec2020ToLMSMat =
+    {
+         0.412109375,     0.52392578125, 0.06396484375,
+         0.166748046875,  0.720458984375, 0.11279296875,
+         0.024169921875,  0.075439453125, 0.900390625
+    };
+
+    return mul(Rec2020ToLMSMat, Rec2020Input);
+}
+
+float3 Rotate709ToLMS(float3 Rec709Input)
+{
+    static const float3x3 Rec709ToLMSMat =
+    {
+         0.412109375,     0.52392578125, 0.06396484375,
+         0.166748046875,  0.720458984375, 0.11279296875,
+         0.024169921875,  0.075439453125, 0.900390625
+    };
+    return mul(Rec709ToLMSMat, Rec709Input);
+}
+
+// Ref: ICtCp Dolby white paper (https://www.dolby.com/us/en/technologies/dolby-vision/ictcp-white-paper.pdf)
+float3 RotatePQLMSToICtCp(float3 LMSInput)
+{
+    static const float3x3 PQLMSToICtCpMat = float3x3(
+        0.5f, 0.5f, 0.0f,
+        1.613769f, -3.323486f, 1.709716f,
+        4.378174f, -4.245605f, -0.1325683f
+        );
+
+    return mul(PQLMSToICtCpMat, LMSInput);
+}
+
+float3 RotateLMSToICtCp(float3 lms)
+{
+    float3 PQLMS = LinearToPQ(max(0.0f, lms));
+    return RotatePQLMSToICtCp(PQLMS);
+}
+
+float3 RotateRec2020ToICtCp(float3 Rec2020)
+{
+    float3 lms = RotateRec2020ToLMS(Rec2020);
+    float3 PQLMS = LinearToPQ(max(0.0f, lms));
+    return RotatePQLMSToICtCp(PQLMS);
+}
+
+
+
+float3 RotateOutputSpaceToICtCp(float3 inputColor)
+{
+    // TODO: Do the conversion directly from Rec709 (bake matrix Rec709 -> XYZ -> LMS)
```

Comment:
guess the compiler may do it?

Reply:
This is a bit different, what I mean is essentially have a version that is RotateRec2020ToICtCp without needing the extra step in between. 

I am not sure the compiler will be able to merge matrices by itself. 

That said, this is not a big deal as it all happen during LUT building.